### ISSUE TITLE
Jina Reranker Models Not Supported in text-embeddings-inference

### ISSUE THREAD

**Issue by arjungandeeva** [huggingface/text-embeddings-inference#554]
### System Info

I tried running the jinaai/jina-reranker-v1-turbo-en model using the text-embeddings-inference container, but it fails due to missing files and an incompatible output format.

### Information

- [x] Docker
- [ ] The CLI directly

### Tasks

- [ ] An officially supported command
- [ ] My own modifications

### Reproduction

docker run -p 8190:80 -v /home/sdp/vignesh/model_weights/tei/:/data --name rerank-jina-sdturbosdtei_svignesh ghcr.io/huggingface/text-embeddings-inference:cpu-1.6 --model-id jinaai/jina-reranker-v1-turbo-en

### Error

2025-04-03T06:07:09.199669Z  INFO text_embeddings_router: router/src/main.rs:185: Args { model_id: "jin***/****-********-**-****o-en", revision: None, tokenization_workers: None, dtype: None, pooling: None, max_concurrent_requests: 512, max_batch_tokens: 16384, max_batch_requests: None, max_client_batch_size: 32, auto_truncate: false, default_prompt_name: None, default_prompt: None, hf_api_token: None, hf_token: None, hostname: "44009ce7e4c1", port: 80, uds_path: "/tmp/text-embeddings-inference-server", huggingface_hub_cache: Some("/data"), payload_limit: 2000000, api_key: None, json_output: false, disable_spans: false, otlp_endpoint: None, otlp_service_name: "text-embeddings-inference.server", cors_allow_origin: None }
2025-04-03T06:07:09.286069Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:20: Starting download
2025-04-03T06:07:09.286087Z  INFO download_artifacts:download_pool_config: text_embeddings_core::download: core/src/download.rs:53: Downloading `1_Pooling/config.json`
2025-04-03T06:07:09.413170Z  WARN download_artifacts: text_embeddings_core::download: core/src/download.rs:26: Download failed: request error: HTTP status client error (404 Not Found) for url (https://huggingface.co/jinaai/jina-reranker-v1-turbo-en/resolve/main/1_Pooling/config.json)
2025-04-03T06:07:10.320323Z  INFO download_artifacts:download_new_st_config: text_embeddings_core::download: core/src/download.rs:77: Downloading `config_sentence_transformers.json`
2025-04-03T06:07:11.327841Z  WARN download_artifacts: text_embeddings_core::download: core/src/download.rs:36: Download failed: request error: HTTP status client error (404 Not Found) for url (https://huggingface.co/jinaai/jina-reranker-v1-turbo-en/resolve/main/config_sentence_transformers.json)
2025-04-03T06:07:11.327853Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:40: Downloading `config.json`
2025-04-03T06:07:11.327885Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:43: Downloading `tokenizer.json`
2025-04-03T06:07:11.327906Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:47: Model artifacts downloaded in 2.041838213s
2025-04-03T06:07:11.327938Z  WARN text_embeddings_router: router/src/lib.rs:403: The `--pooling` arg is not set and we could not find a pooling configuration (`1_Pooling/config.json`) for this model but the model is a BERT variant. Defaulting to `CLS` pooling.
2025-04-03T06:07:11.393769Z  WARN text_embeddings_router: router/src/lib.rs:188: Could not find a Sentence Transformers config
2025-04-03T06:07:11.393778Z  INFO text_embeddings_router: router/src/lib.rs:192: Maximum number of tokens per request: 8192
2025-04-03T06:07:11.393886Z  INFO text_embeddings_core::tokenization: core/src/tokenization.rs:28: Starting 152 tokenization workers
2025-04-03T06:07:12.694237Z  INFO text_embeddings_router: router/src/lib.rs:234: Starting model backend
2025-04-03T06:07:12.694341Z  INFO text_embeddings_backend: backends/src/lib.rs:534: Downloading `model.onnx`
2025-04-03T06:07:12.982703Z  WARN text_embeddings_backend: backends/src/lib.rs:538: Could not download `model.onnx`: request error: HTTP status client error (404 Not Found) for url (https://huggingface.co/jinaai/jina-reranker-v1-turbo-en/resolve/main/model.onnx)
2025-04-03T06:07:12.982716Z  INFO text_embeddings_backend: backends/src/lib.rs:539: Downloading `onnx/model.onnx`
2025-04-03T06:07:15.709508Z  INFO text_embeddings_backend: backends/src/lib.rs:548: Downloading `model.onnx_data`
2025-04-03T06:07:15.823268Z  WARN text_embeddings_backend: backends/src/lib.rs:552: Could not download `model.onnx_data`: request error: HTTP status client error (404 Not Found) for url (https://huggingface.co/jinaai/jina-reranker-v1-turbo-en/resolve/main/model.onnx_data)
2025-04-03T06:07:15.823275Z  INFO text_embeddings_backend: backends/src/lib.rs:553: Downloading `onnx/model.onnx_data`
2025-04-03T06:07:15.949514Z  WARN text_embeddings_backend: backends/src/lib.rs:557: Could not download `onnx/model.onnx_data`: request error: HTTP status client error (404 Not Found) for url (https://huggingface.co/jinaai/jina-reranker-v1-turbo-en/resolve/main/onnx/model.onnx_data)
2025-04-03T06:07:15.949522Z  INFO text_embeddings_backend: backends/src/lib.rs:349: Model ONNX weights downloaded in 3.255183115s
Error: Model backend is not healthy

Caused by:
    Unknown output keys: [Output { name: "logits", output_type: Tensor { ty: Float32, dimensions: [-1, 1], dimension_symbols: [Some("batch_size"), None] } }]

**Comment by CoolFish88** [comment]
Facing a similar issue when deploying on a Sagemaker GPU endpoint (ml.g5.2xlarge):

"Error: The `--pooling` arg is not set and we could not find a pooling configuration (`1_Pooling/config.json`) for this model."

**Comment by alvarobartt** [comment]
Hey @arjungandeeva thanks for reporting, could you try with the latest version being 1.6.1? I just tried to reproduce and it worked fine on my end! Thanks in advance!

**Comment by arjungandeeva** [comment]
@alvarobartt Still the same issue with 1.6.1. COuld you please share the command that you are trying and hardware configuration?
Iam using the cpu image.

**Comment by alvarobartt** [comment]
Hey @arjungandeeva so the issue won't happen on MPS, but will happen on CUDA and ONNX, so I'll investigate it and come back to you once fixed, but is indeed not working fine on those as reported originally in your issue! Sorry if the message above caused any confusion!

**Comment by CoolFish88** [comment]
+1 for this initiative. Thanks @alvarobartt for the hard work on many fronts, this one included.

I am also interested in using jinaai/jina-reranker-v1-turbo-en with TEI for deployment on Sagemaker GPU instances.

**Comment by alvarobartt** [comment]
Hey @CoolFish88, thanks for the kind words, it's pretty much a team effort indeed! As a heads up, just tried `jinaai/jina-reranker-v1-turbo-en` with the latest release and it seems to be working fine on both MPS and CUDA, but the ONNX weights are failing, I'll review those in detail and report back!

**Comment by alvarobartt** [comment]
Hey again @arjungandeeva and @CoolFish88, until it's merged you can use the revision `refs/pr/13` that contains the fix required to make it work with Text Embeddings Inference (TEI), as per https://huggingface.co/jinaai/jina-reranker-v1-turbo-en/discussions/13. 🤗

**Comment by CoolFish88** [comment]
Hello @alvarobartt,

I created a new issue documenting some errors related to jina rerankers deployment: https://github.com/huggingface/text-embeddings-inference/issues/579. 

I tried the approach you listed here but encountered a new error down the line
